cinema/views.py

```diff
@@ -1 +1,49 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, MovieSession, Actor, Genre, CinemaHall
+from cinema.serializers import (MovieListSerializer,
+                                MovieSerializer,
+                                MovieDetailSerializer,
+                                MovieSessionSerializer,
+                                MovieSessionListSerializer,
+                                MovieSessionDetailSerializer,
+                                ActorSerializer,
+                                GenreSerializer,
+                                CinemaHallSerializer)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("genres", "actors")
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.prefetch_related("cinema_hall")
```

Comment:
does moviesession has m2m relations ? also the same about optimizing query as above